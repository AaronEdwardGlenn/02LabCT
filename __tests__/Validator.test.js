const Validator = require('../lib/Validator.js');

describe('Validator', () => {
    let validator; 

    describe('required fields', () => {
        beforeAll(function(){
            validator = new Validator('age', {
                type: Number, 
                required: true
            });
        });

        it('correctly returns the field that was alwasy correct. for realz', () => {
            const dog = {
                name: 'scruffy', 
                age: 69, 
                weight: '420 lbs'
            };

            expect(validator.validate(dog)).toEqual(69);
        });

        it('returns the field that was not input correctly and cast that field to type', () => {
            const dog = {
                name: 'scrapppp', 
                age: '69', 
                weight: '420 lbs', 
            }; 
            expect(validator.validate(dog)).toEqual(69);
        });

        it ('field does not exist and was required so explotion and error thrown', () => {
            const dog = {
                name: 'scruffy', 
                weight: '420 lbs'
            };

            expect(() => validator.validate(dog)).toThrowErrorMatchingSnapshot(); 
        });
    }); 

    describe('optional fields', () => {
        beforeAll(() => {
            validator = new Validator('age', {
                type: Number
            }); 
        }); 

        it('correctly returns the field that was alwasy correct. for realz', () => {
            const dog = {
                name: 'scrapps',
                age: 69, 
                weight: '420 lbs'
            };

            expect(validator.validate(dog)).toEqual(69);

        });

        it('returns the field that was not input correctly and cast that field to type', () => {
            const dog = {
                name: 'scruffffffffff', 
                age: '69', 
                weight: '420 lbs'
            };

            expect(validator.validate(dog)).toEqual(69); 
        }); 

        it('field does not exist and was not required so null is fine', () => {
            const dog = {
                name: 'scrappers', 
                weight: '420 lbs'
            }; 

            expect(validator.validate(dog)).toBeNull(); 
        }); 
    });
});