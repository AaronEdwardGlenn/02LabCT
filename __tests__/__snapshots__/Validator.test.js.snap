// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator required fields field does not exist and was required so explotion and error thrown 1`] = `"age needs to be here. What are you doing. Serously."`;
